posthog/async_migrations/migrations/0002_fill_pdi2_from_pdi.py

```diff
@@ -0,0 +1,140 @@
+from constance import config
+from django.conf import settings
+
+from ee.clickhouse.client import sync_execute
+from ee.clickhouse.sql.events import EVENTS_TABLE
+from posthog.async_migrations.definition import AsyncMigrationDefinition, AsyncMigrationOperation
+from posthog.constants import AnalyticsDBMS
+from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_DATABASE
+from posthog.version_requirement import ServiceVersionRequirement
+
+
+class Migration(AsyncMigrationDefinition):
+
+    description = """
+        Schema change to migrate the data from the old person_distinct_id table
+        to the new person_distinct_id2 table. This is required as the existing
+        person_distinct_id does not use a strictly increasing id for
+        identifying the latest (distinct_id, person_id) pairing.
```

Comment:
This reason is wrong - it was possible to correctly query it, just it was really slow for our largest users. The versioning logic is just a cherry on top to simplify things.

Reply:
So is the assumption that you can't undelete a person and thus if there is a delete we know it must be the last row? It wasn't clear to me if this was the case but yup if that is the case I guess we don't care so much.